### PR TITLE
Add utility function to get image repo and tag from image reference

### PR DESCRIPTION
**Pull request by danielfoehrKn**
**What this PR does / why we need it**:

Adding a simple utility function to get image repo and tag from an image reference.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

**Comment by danielfoehrKn** [comment]
@schrodit

File: bindings-go/apis/v2/cdutils/resources.go [hunk 1]
```diff
@@ -42,3 +43,20 @@ func GetImageReferenceByName(cd *cdv2.ComponentDescriptor, name string) (string,
...
+// e.g eu.gcr.io/gardener-project/gardener/gardenlet:v1.11.3
+// and returns the image repository as a first, and the tag as a second argument
+func GetRepositoryAndTagFromReference(imageReference string) (string, string, error) {
+	split := strings.Split(imageReference, ":")
+	if len(split) == 2 {
+		return split[0], split[1], nil
+	}
+
+	// split version from reference if image reference contains a port
+	// e.g eu.gcr.io/gardener-project:5000/gardener/gardenlet
```
Comment (schrodit) [inline bindings-go/apis/v2/cdutils/resources.go:57]:
```suggestion
	// e.g eu.gcr.io:5000/gardener-project/gardener/gardenlet
```

File: bindings-go/apis/v2/cdutils/resources.go [hunk 2]
```diff
@@ -42,3 +43,20 @@ func GetImageReferenceByName(cd *cdv2.ComponentDescriptor, name string) (string,
 	}
 	return ociImageAccess.ImageReference, nil
 }
+
+// GetRepositoryAndTagFromReference takes an image reference
+// e.g eu.gcr.io/gardener-project/gardener/gardenlet:v1.11.3
+// and returns the image repository as a first, and the tag as a second argument
+func GetRepositoryAndTagFromReference(imageReference string) (string, string, error) {
```
Comment (schrodit) [inline bindings-go/apis/v2/cdutils/resources.go:50]: you should also consider hashes liek `eu.gc.rio/gardener-project/gardenlet@sha256:12345`

**Review by schrodit — CHANGES_REQUESTED** [review]

**Review by schrodit — APPROVED** [review]
/lgtm
looks good now

**Pull request merged**